packages/router/src/bindings/contractReader/index.ts

```diff
@@ -72,21 +84,41 @@ export const handleActiveTransactions = async (transactions: ActiveTransaction<a
       transaction.crosschainTx.invariant.transactionId,
     );
     if (handlingTracker.has(transaction.crosschainTx.invariant.transactionId)) {
-      logger.info("Already handling transaction", requestContext, methodContext);
+      logger.debug("Already handling transaction", requestContext, methodContext);
       continue;
     }
-    handlingTracker.set(transaction.crosschainTx.invariant.transactionId, transaction.status);
-    handleSingle(transaction, requestContext).finally(() =>
-      handlingTracker.delete(transaction.crosschainTx.invariant.transactionId),
-    );
+
+    let chainId: number;
+    if (
+      transaction.status === CrosschainTransactionStatus.SenderPrepared ||
+      transaction.status === CrosschainTransactionStatus.SenderExpired
+    ) {
+      chainId = transaction.crosschainTx.invariant.receivingChainId;
+    } else {
+      chainId = transaction.crosschainTx.invariant.sendingChainId;
+    }
+
+    handlingTracker.set(transaction.crosschainTx.invariant.transactionId, {
+      blockNumber: 0,
+      chainId,
+    });
+    const res = await handleSingle(transaction, requestContext);
+    if (res) {
+      handlingTracker.set(transaction.crosschainTx.invariant.transactionId, {
+        blockNumber: res.blockNumber,
+        chainId,
+      });
+    } else {
+      handlingTracker.delete(transaction.crosschainTx.invariant.transactionId);
```

Comment:
Why are we delaying when there isn't a `res` ?

Reply:
we are delaying but deleting the record if res is undefined.

Reply:
I meant deleting lol

Reply:
What does `res` being undefined mean exactly?

Reply:
handleSingle resolves with undefined if it errors or transaction fails